packages/desktop/src/renderer/components/hooks/useInput.ts

```diff
@@ -0,0 +1,11 @@
+import { useEngineStore } from '@renderer/engine'
+
+export const useInput = (id: string) => {
+  const inputs = useEngineStore((state) => state.inputs)
+  return inputs[id]
+}
```

Comment:
Some performance things to consider here. `useEngineStore` will cause a component re-render if it is returning a new object compared to the last render.  Whenever an input changes (e.g. elsewhere in the app an input is added), the `inputs` object will update, causing `useEngineStore` to trigger a re-render. You can fix this by directly selecting inside `useEngineStore`. You don't even need to have this in a separate hook if you don't want to, it's just nice sometimes to keep this abstracted away (especially if it then becomes more complex later on).

```suggestion
export const useInput = (id: string) => {
  return useEngineStore((state) => state.inputs[id])
}
```